sklearn/datasets/_svmlight_format_io.py

```diff
@@ -393,39 +380,34 @@ def _dump_svmlight(X, y, f, multilabel, one_based, comment, query_id):
 
         f.write(b"#\n")
         f.writelines(b"# %s\n" % line for line in comment.splitlines())
-
-    for i in range(X.shape[0]):
-        if X_is_sp:
-            span = slice(X.indptr[i], X.indptr[i + 1])
-            row = zip(X.indices[span], X.data[span])
-        else:
-            nz = X[i] != 0
-            row = zip(np.where(nz)[0], X[i, nz])
-
-        s = " ".join(value_pattern % (j + one_based, x) for j, x in row)
```

Comment:
In general, I think that `join` is faster than using string concatenation (which is being used in this PR)

```python
def append_string():
    my_str = ""
    for i in range(100):
        my_str += f"{i},"
    return my_str

def join_string():
    return ",".join(f"{i}" for i in range(100))

%timeit append_string()
10.2 µs ± 58.5 ns per loop

%timeit join_string()
6.34 µs ± 60.6 ns per loop
```

(These are not exactly the same)